solara/lab/components/confirmation_dialog.py

```diff
@@ -0,0 +1,49 @@
+from typing import cast, Callable, List, Union, Optional
+
+import solara
+import reacton.ipyvuetify as v
+
+
+@solara.component
+def ConfirmationDialog(
+    is_open_rv: solara.Reactive[bool],
+    action: Callable[[], None],
+    content: Union[str, solara.Element] = "",
+    title: str = "Confirm action",
+    on_close: Callable[[], None] = lambda: None,
+    ok: Union[str, solara.Element] = "OK",
+    cancel: Union[str, solara.Element] = "Cancel",
+    children: List[solara.Element] = [],
+):
+    def perform_action():
+        action()
+        close()
+
+    def close():
+        on_close()  # possible additional actions when closing
+        is_open_rv.set(False)
+
+    if is_open_rv.value:
+        with v.Dialog(
+            #children=children,
+            v_model=is_open_rv.value,
+            on_v_model=is_open_rv.set,
+            permanent=True,
+            max_width=500,
+        ):
+            with solara.Card(title=title):
+                if isinstance(content, str):
+                    solara.Markdown(content)
+                else:
+                    solara.display(content)
+                if children:
+                    solara.display(*children)
+                with solara.CardActions():
+                    if isinstance(ok, str):
+                        solara.Button(label=ok, on_click=perform_action)
+                    else:
+                        solara.display(ok)
```

Comment:
I see an issue here is that a user provided button will not trigger perform_action. It would be nice to expose this during the tests. It might be that if we need to use `use_event` for this that we hit this bug: https://github.com/widgetti/reacton/issues/14

Reply:
Yes, as I mentioned in our call, if you write a test we can find a solution to this later on, I have some ideas about it.

Reply:
In theory, we should be able to use `solara.v.use_event`, but it has issues due to https://github.com/widgetti/reacton/issues/14

I think we can do a different approach, which is to inspect ok.kwargs["on_click"] and replace it by a new dict where on_click points to our perform_action, and the perform_action also calls the previous `on_click` set by the user.

Reply:
So, if I understand this correctly, if you pass a custom `Button` named `ok`, then `ok.kwargs['on_click']` is only present if you passed an `on_click` event to your button. Should we allow for a case where a user passes in a custom button that has *no* `on_click`? Perhaps because they just want to execute `on_ok` as intended.

Related: what do we do in the case that both `on_ok` and a custom button's `on_click` are set?

Reply:
For context, I currently have this:

```python
    with solara.CardActions():
        if isinstance(ok, str):
            solara.Button(label=ok, on_click=perform_action)
        else:
            solara.display(ok)
            if ok.kwargs.get('on_click'):
                ok.kwargs['on_click'] = perform_action
```

...which passes the test. I think there is no point in checking `ok.kwargs` if `ok` is a string?

Reply:
Yes, I think we should support no on_click as well. I suggest we also don't modify the dict, but create a new one.

Reply:
> In theory, we should be able to use `solara.v.use_event`, but it has issues due to [widgetti/reacton#14](https://github.com/widgetti/reacton/issues/14)
> 
> I think we can do a different approach, which is to inspect ok.kwargs["on_click"] and replace it by a new dict where on_click points to our perform_action, and the perform_action also calls the previous `on_click` set by the user.

I should have read this more carefully, you're literally describing what to do in a situation with both `on_ok` and `on_click` being present.